### ISSUE TITLE
Doc request: How to filter kamon jars from packaged artifact

### ISSUE THREAD

**Issue by paddymahoney** [kamon-io/Kamon#179]
Hey Kamon-folk, 

I find the kamon ask timeout tracing quite useful in develop, but even with
kamon.trace.ask-pattern-tracing = off
there is a cost, and I would like to avoid that in prod.

I'd like to include the jars as a dependency only in Compile, Test, but would like to filter/remove them in packaging. We use sbt-native-packager for generating our packages.

Feel free to close if you feel this isn't widely-enough applicable to document.

Thanks for your work on Kamon Ivan and Diego!

**Comment by paddymahoney** [comment]
oh shoot, I forgot the code. Here is what I use to exclude the kamon jars:
```
object Kamon {

  import com.typesafe.sbt.SbtNativePackager.Universal

  val version = "0.3.5"

  val settings = Seq(
    libraryDependencies += "io.kamon" %% "kamon-core" % version)

  val packagingSettings = Seq(
    mappings in Universal ~= (_.filterNot {case (file, _) => file.getName.contains("kamon")}))
}
```

I'm currently testing to see whether not including the weaving argument is actually enough.

**Comment by ivantopo** [comment]
hello @paddymahoney, hope you are doing fine! This is certainly the first time we see a request like this, typically people want to use Kamon in production to monitor their stuff but yes, as you said, there is a cost on doing so, specially when the ask pattern as it creates an exception every time you ask just to grab the stacktrace. In the newest version we enhanced the instrumentation as explained in #113, have you tried the new `lightweight` mode?

**Comment by ivantopo** [comment]
Patrick, I'll proceed to close this issue as it certainly seems to be a very isolated case.. still, thanks for sharing your solution to filtering artifacts, that will remain here for anyone that might have a similar requirement in the future!

**Comment by paddymahoney** [comment]
Just wanted to add that in the end I ended up leaving the artifacts - because the javaagent isn't set to include the aspectj weaver in our prod system, there don't seem to be any performance impacts to speak of. 

Question re: the lightweight mode - is that in 0.3.6? 

Thanks for your help Ivan.

**Comment by ivantopo** [comment]
Nice, if you keep the jars there but no agent it will definitely not change a single piece of bytecode and no overhead is introduced, still you get the chance to use Kamon for non-akka-specific metrics, in case you want to track some other stuff.

On other note, the performance impact was specific to heavy use of the ask pattern or was it a general slowdown of your app?

And yes, that lightweight mode is included in the 0.3.6 snapshots, which will become 0.4.0 pretty soon. Thanks again for sharing!
